pkg/backup/file_sst_sink.go

```diff
@@ -131,10 +138,20 @@ func (s *fileSSTSink) flushFile(ctx context.Context) error {
 		return nil
 	}
 
+	var lastFile *backuppb.BackupManifest_File
```

Comment:
Ideally, this change doesn't add further complexity to the existing file sink implementation. What if you created a new struct that wrapped the fileSSTSink? i.e.

```
type fileSinkKeyWriter struct

  fileSSTSink

  prevKey roachpb.Key
```

with the main methods `WriteKey` and `Flush`. For flush:
```
(s *fileSinkKeyWriter) Flush() {
  if err := s.setMidRowForPrevKey(lastFile.Span.EndKey); err != nil {
		return err
	}
  // call the underlying file sink's flush
  s.flush()
  
}
```

I'm also wondering if we should wrap the file sink in some interface, but i'm not sure the best way to do that. Going to ask in -eng. That shouldn't block this suggestion though.